File: my-solutions/0643-maximum-average-subarray-i/solution.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int max_sum = 0;
        int current_sum = 0;
        
        for (int i = 0; i < k; ++i) {
            current_sum += nums[i];
        }
        max_sum = current_sum;
        
        for (int i = k; i < nums.size(); ++i) {
            current_sum += nums[i];
            current_sum -= nums[i - k];
            max_sum = max(current_sum, max_sum);
        }
        
        return max_sum / double(k);
    }
};

File: my-solutions/0643-maximum-average-subarray-i/solution.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        running = 0
        for i in range(k):
            running += nums[i]
        maxAvg = running / k
        
        start, end = 0, k
        while end < len(nums):
            running -= nums[start]
            start += 1
            running += nums[end]
            maxAvg = max(maxAvg, running / k)
            end += 1
        
        return maxAvg
